### ISSUE TITLE
Python can't call torrent_info.map_file()

### ISSUE THREAD

**Issue by DjLegolas** [arvidn/libtorrent#3780]
libtorrent version: 1.2.0

platform/architecture: windows 10, x86

compiler and compiler version: boost_1_69_0 and Visual Studio 2017 Enterprise.

Trying to call torrent_info.map_file(0, 0, 0) and receiving a TypeError:
```
E       TypeError: No Python class registered for C++ class struct libtorrent::aux::strong_typedef<int,struct libtorrent::aux::piece_index_tag,void>
```
It works with libtorrent_1_1_12 and the same boost version.

Using python 2.7.

**Comment by arvidn** [comment]
there is a call to this function in `test.py`, which passes on all platforms including windows. Are you building with deprecated functions disabled by any chance?

**Comment by arvidn** [comment]
I tried on OSX (with boost-1.70, python3, both with and without deprecated functions). I cannot reproduce this with the test program, `test.py`. Does that `test.py` fail for you as well?

**Comment by DjLegolas** [comment]
Tried with both boost-1.69 and boost-1.70, and failed both in 4 different tests:
```
======================================================================
ERROR: test_from_scratch (__main__.test_create_torrent)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test.py", line 51, in test_from_scratch
    ct.add_url_seed('foo')
ArgumentError: Python argument types in
    create_torrent.add_url_seed(create_torrent, str)
did not match C++ signature:
    add_url_seed(struct libtorrent::create_torrent {lvalue}, class boost::basic_string_view<char,struct std::char_traits<char> >)

======================================================================
ERROR: test_construct (__main__.test_dht_settings)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test.py", line 757, in test_construct
    ds = lt.dht_settings()
AttributeError: 'module' object has no attribute 'dht_settings'

======================================================================
ERROR: test_peer_info_members (__main__.test_peer_info)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test.py", line 749, in test_peer_info_members
    print(p.read_state)
TypeError: No Python class registered for C++ class struct libtorrent::flags::bitfield_flag<unsigned char,struct libtorrent::bandwidth_state_flags_tag,void>

======================================================================
ERROR: test_torrent_handle (__main__.test_torrent_handle)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test.py", line 112, in test_torrent_handle
    print(torrent_files.map_file(0, 0, 0).piece)
TypeError: No Python class registered for C++ class struct libtorrent::aux::strong_typedef<int,struct libtorrent::aux::piece_index_tag,void>

----------------------------------------------------------------------
```

Full test run can be found [here](https://pastebin.com/NGGUrSsU).

**Comment by arvidn** [comment]
how are you building libtorrent and the python bindings?

**Comment by arvidn** [comment]
actually, I tested with the `RC_1_2` branch. it's possible I've fixed this since the release of 1.2, for the next release

**Comment by arvidn** [comment]
I just tested the release tag for 1.2, and it works for me as well (this time on Linux, python 2.7, boost-1.68.0)

**Comment by DjLegolas** [comment]
> how are you building libtorrent and the python bindings?

i'm using the setup.py file, in `binding/python` (as suggested in the [docs](https://github.com/arvidn/libtorrent/blob/RC_1_2/docs/python_binding.rst)), using VS2017 Enterprise.
> I just tested the release tag for 1.2, and it works for me as well (this time on Linux, python 2.7, boost-1.68.0)

and for me on Windows 10, python 2.7, boost-1.69.0, and failed again.
> actually, I tested with the RC_1_2 branch. it's possible I've fixed this since the release of 1.2, for the next release

well, i built lt-1.2 again (boost-1.70.0) using the latest `RC_1_2` branch, and all of the tests passed.

seem like a problem with windows...
